### DISCUSSION TITLE
How to plot only few points in a dataframe.

### DISCUSSION THREAD

**Discussion by bhawesh-bks** [pyvista/pyvista#3306] [Q&A] [answered]
So I have a csv file which has x,y,z coordinates and DT_pred(which is predicted value). I want to plot a 3d plot with only specific values from the the DT_pred. But whenever i try to plot it, it either gives me only the surface plot or else plot as mentioned below.
![Screenshot (14)](https://user-images.githubusercontent.com/97894832/189821095-6ef80221-7d3a-41e0-b974-9ffbad568e1f.png)
I just want the volume plot of the area which is termed as True. 
In the above plot i haven taken the value of Dt_pred to be 1.
I am also attaching the csv file 
[raw_data.csv](https://github.com/pyvista/pyvista/files/9553936/raw_data.csv)
Is there a way to plot like the image given below.
![minerals-08-00443-g008](https://user-images.githubusercontent.com/97894832/189823248-99ff73de-5cb4-4a4a-80ed-f0f82751741e.png)

**Comment by banesullivan** [comment, marked as answer]
Where did you get these data? Is it possible to extract the data in a more useable format? This CSV is just the nodes of an original grid/mesh. Looking at this mesh in full, you can see a regular structure. This means we could do some hackery to attempt to reconstruct the gird. For example:

```py
import pyvista as pv
import pandas as pd
import numpy as np

df = pd.read_csv('raw_data.csv')

pts = df[['x', 'y', 'z']].values
dt = df['DT_pred'].values


poly = pv.PolyData(pts)
poly['DT_pred'] = dt

poly.plot(notebook=0, color=True, show_grid=True)
```

![Screen Shot 2022-09-13 at 10 09 12 AM](https://user-images.githubusercontent.com/22067021/189952222-2866f2ce-27d8-4198-82dc-dc0f88dc5509.png)



```py
>>> len(np.unique(pts[:,0])), len(np.unique(pts[:,1]))
(89, 119)
```

The z coordinate will be whatever is left over. I can assume/trust that the X and Y coordinates are a regular lattice and thus recover the dimensions of the grid.

```py
>>> len(pts) / (89 * 119)
40.0
```

The fact that worked out to a whole number gives me confidence that the dimensions of the lattice are `(89, 119, 40)`. So we can generate a `pyvista.StructuredGrid` of these points with those dimensions:

```py
grid = pv.StructuredGrid()
grid.points = pts
grid.dimensions = (89, 119, 40)
# Add scalar values
grid['DT_pred'] = dt

# Display as a volumetric grid!
grid.plot(notebook=0, show_edges=True)
```

![Screen Shot 2022-09-13 at 10 14 10 AM](https://user-images.githubusercontent.com/22067021/189953329-fb5b9e61-5d4a-44bc-af35-990a49a718ef.png)


Now we can go in an extract isovolumes like you show above!

```py
iso = grid.threshold([1.5, 2.5], scalars='DT_pred')
iso.plot(notebook=0)
```
![Screen Shot 2022-09-13 at 10 15 59 AM](https://user-images.githubusercontent.com/22067021/189953607-5b47421d-4d57-4faf-96ab-813e12a33d94.png)


But that doesn't look to great, so let's try contouring:

```py
cntrs = grid.contour([2.0,], scalars='DT_pred')
cntrs.plot(notebook=0, color=True)
```
![Screen Shot 2022-09-13 at 10 17 47 AM](https://user-images.githubusercontent.com/22067021/189954087-6b5d2e05-55be-418b-b7ad-756d9c6f07e9.png)


Eh... still not amazing looking but hopefully this gets you started and having a bit more insight into the actual data, you can get this right.

FYI, you may be interested in the smoothing filter: https://docs.pyvista.org/examples/01-filter/surface-smoothing.html

**Reply by bhawesh-bks** [reply]
Thanks a lot @banesullivan . That helps a lot
